main_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDateParamToEpoch(t *testing.T) {
+
+	timeNow = func() time.Time {
+		//16 Aug 1994 15:30
+		return time.Date(1994, time.August, 16, 15, 30, 0, 100, defaultTimeZone)
+	}
+
+	const shortForm = "15:04 2006-Jan-02"
+
+	var tests = []struct {
+		input  string
+		output string
+	}{
+		{"midnight", "00:00 1994-Aug-17"},
+		{"noon", "12:00 1994-Aug-16"},
+		{"teatime", "16:00 1994-Aug-16"},
+
+		{"noon 08/16/94", "12:00 1994-Aug-16"},
+		{"midnight 20060816", "00:00 2006-Aug-16"},
+
+		{"15:04 19940816", "15:04 1994-Aug-16"},
+		{"-1day", "15:30 1994-Aug-15"},
+	}
+
+	for _, tt := range tests {
+		parsedTime := dateParamToEpoch(tt.input, int64(0))
```

Comment:
s/parsedTime/got/   (see below)
The `0` is a untyped constant, which means you don't need to convert it to an `int64`.  The compiler will do that for you.

https://blog.golang.org/constants